src/Pages/Templates/TemplateDetails/components/AddSystems/ModalSystemsTable.tsx

```diff
@@ -0,0 +1,168 @@
+  allSelected: boolean;
+}
+
+export default function ModalSystemsTable({
+  isFetchingOrLoading,
+  isLoadingOrZeroCount,
+  systemsList,
+  perPage,
+  sortParams,
+  setSelected = () => undefined,
+  selected,
+  selectAllToggle,
+  allSelected,
+}: Props) {
+  const classes = useStyles();
+  const [prevLength, setPrev] = useState(perPage || 10);
+  const rootPath = useRootPath().slice(1); // Using hrefs, we need to remove
+  const { templateUUID: uuid = '' } = useParams();
+  const basePath = rootPath.replace('content', '');
+
+  useEffect(() => {
+    setPrev(systemsList.length || 10);
+  }, [systemsList.length]);
+
+  const columnHeaders = ['Name', 'Tags', 'OS', 'Template'];
+
+  return (
+    <>
+      <Hide hide={!isFetchingOrLoading}>
+        <Grid className={classes.mainContainer}>
+          <SkeletonTable
+            rows={prevLength}
+            numberOfColumns={columnHeaders.length}
+            variant={TableVariant.compact}
+          />
+        </Grid>
+      </Hide>
+      <Hide hide={isFetchingOrLoading}>
+        <Table aria-label='assign systems table' ouiaId='assign_systems_table' variant='compact'>
+          <Hide hide={isLoadingOrZeroCount}>
+            <Thead>
+              <Tr>
+                <Th
+                  className={classes.fixLabelFontSize}
+                  aria-label='select-all-box'
+                  select={{
+                    onSelect: () => selectAllToggle(),
+                    isSelected: allSelected,
+                  }}
+                />
+                {columnHeaders.map((columnHeader, index) => (
+                  <Th key={columnHeader + '_column'} sort={sortParams && sortParams(index)}>
+                    {columnHeader}
+                  </Th>
+                ))}
+              </Tr>
+            </Thead>
+          </Hide>
+          <Tbody>
+            {systemsList.map(
+              (
+                { id, attributes: { display_name, tags, os, template_name, template_uuid } },
+                rowIndex,
+              ) => (
+                <Tr key={id + rowIndex}>
+                  <TdWithTooltip
+                    className={classes.fixLabelFontSize}
+                    show={template_uuid === uuid}
+                    tooltipProps={{
+                      content: `System '${display_name}' is already assigned to template '${template_name}'.`,
+                      position: TooltipPosition.right,
+                    }}
+                    select={{
+                      rowIndex,
+                      isDisabled: template_uuid === uuid,
+                      onSelect: () => setSelected(id),
+                      isSelected: selected.has(id) || template_uuid === uuid,
+                    }}
+                  />
+                  <Td>
+                    <Button
+                      isInline
+                      variant='link'
+                      component='a'
+                      href={`${basePath}${PATCH_SYSTEMS_ROUTE}${id}`}
```

Comment:
do you think we should route to the systems and the template details from this modal in a different tab so the modal is kept open? just curious, i'm not sure if there is a benefit or if it's more disorienting to the user to open a new tab

Reply:
The standard we have been following is to allow the user to middle click if they want to open another tab. 
I can ask Maria about this on Wednesday.
We don't normally default to opening another tab for a user on click, but this might be a good pattern to switch to.

Reply:
I've added this as I think it makes sense. 
We can change it back and review this functionality if necessary after our chat with design.